packages/neon/neon/lib/src/blocs/push_notifications.dart

```diff
@@ -32,16 +32,22 @@ class PushNotificationsBloc extends Bloc implements PushNotificationsBlocEvents,
     if (_platform.canUsePushNotifications) {
       unawaited(UnifiedPush.getDistributors().then(_globalOptions.updateDistributors));
 
-      _globalOptions.pushNotificationsEnabled.stream.listen((final enabled) async {
-        if (enabled != _pushNotificationsEnabled) {
-          _pushNotificationsEnabled = enabled;
-          if (enabled) {
-            // We just use a single RSA keypair for all accounts
-            _keypair = await PushUtils.loadRSAKeypair(_storage);
-            await _setupUnifiedPush();
-          }
-        }
-      });
+      _globalOptions.pushNotificationsEnabled.addListener(_pushNotificationsEnabledListener);
+    }
+  }
+
+  Future<void> _pushNotificationsEnabledListener() async {
+    final enabled = _globalOptions.pushNotificationsEnabled.value;
+    if (enabled != _pushNotificationsEnabled) {
+      _pushNotificationsEnabled = enabled;
+      if (enabled) {
+        // We just use a single RSA keypair for all accounts
+        _keypair = await PushUtils.loadRSAKeypair(_storage);
+        await _setupUnifiedPush();
+      } else {
+        _globalOptions.pushNotificationsDistributor.removeListener(_distributorListener);
+        unawaited(_accountsListener.cancel());
```

Comment:
You cancel the listeners here, but what happens if the user enables push notifications again? As far as I can tell the listeners are only created in initState

Reply:
I haven't tested this (as you mentioned some thorough testing might be needed).
If enabled `_setupUnifiedPush` will be called two lines above wich will register the listener.

I'll have a look we can clean that up/add comments in the code to make that clear.

Reply:
I think I found a nice way to clean it up a bit.

I also removed some `cached` variables as it's now possible to synchronously access the real value removing the possibility of potentially outdated values.